### PR TITLE
Sort interface materials

### LINKED ISSUE #17171: Check InterfaceMaterial dependence resolution
## Bug Description
I don't know if this is bug or not, but we don't have tests for checking `InterfaceMaterial` dependence resolution.
So appropriate tests should be added

## Steps to Reproduce
Example about what was trying to do and didn't worked out properly (possibly because of this issue):
InterfaceKernel consumes property from `interfacematerial1`
`interfacematerial1` needs update properties from `interfacematerial2` and `interfacematerial3`
`interfacematerial2`  needs update properties from `interfacematerial3`
`interfacematerial3` use a coupled variables to generate the required properties.

The execution order was the following at every step (printed out from screen).
interfacematerial2::computeQpProperties()
interfacematerial3::computeQpProperties()
interfacematerial1::computeQpProperties()

## Impact
Prevents from designing more modular code.

@lindsayad is already aware of this, as per our conversation on slack.

### PR DESCRIPTION
**Pull request by lindsayad**
closes #17171

@arovinelli can you see if this resolves your issue?

### CODE REVIEW

**Comment by lindsayad** [comment]
Hmm this doesn't appear to be working on a simple test I drew up...

**Comment by arovinelli** [comment]
@lindsayad for some reason it works on my case. Now the execution order is correct

**Comment by lindsayad** [comment]
Yea I realized what the problem was. I am using AD material properties and we didn't have the correct methods for AD material properties in `InterfacMaterial` yet

**Comment by lindsayad** [comment]
Thanks for reporting this issue!

**Comment by arovinelli** [comment]
> Thanks for reporting this issue!

You are welcome, in the next few days I'll also try with stateful properties. If anything comes up I'll let you know :)

File: framework/doc/content/syntax/Materials/index.md [hunk 1]
```diff
@@ -197,6 +197,14 @@ table lists the types of properties that are available for automatic output.
 | RealVectorValue | `MaterialRealVectorValueAux` | prop_1, prop_2, and prop_3 |
 | RealTensorValue | `MaterialRealTensorValueAux` | prop_11, prop_12, prop_13, prop_21, etc. |
 
+## Material sorting
+
+Materials are sorted such that one material may consume a property produced by
+another material and know that the consumed property will be up-to-date,
+e.g. the producer material will execute before the consumer material. If a
+cyclic dependcy is detected between two materials, then MOOSE will produce an
```
Comment (GiudGiud) [inline framework/doc/content/syntax/Materials/index.md:205]:
```suggestion
cyclic dependency is detected between two materials, then MOOSE will produce an
```

**Review by GiudGiud — COMMENTED** [review]
lgtm

**Review by rwcarlsen — DISMISSED** [review]

**Comment by lindsayad** [comment]
@rwcarlsen could you re-approve after I applied @GiudGiud 's suggestion?

**Review by rwcarlsen — APPROVED** [review]

**Pull request merged**